phoenix-core/src/main/java/org/apache/phoenix/compile/UpsertCompiler.java

```diff
@@ -156,7 +157,7 @@ private static MutationState upsertSelect(PhoenixStatement statement,
                             column.getMaxLength(), column.getScale(), column.getSortOrder());
                     values[i] = ByteUtil.copyKeyBytesIfNecessary(ptr);
                 }
-                setValues(values, pkSlotIndexes, columnIndexes, table, mutation);
+                setValues(values, pkSlotIndexes, columnIndexes, table, mutation, statement);
```

Comment:
Note that this statement may have been cloned from the original statement with a clone of the original connection since this upsertSelect is called in parallel by multiple threads in a upsert select statement. Is that going to be ok for linking back to the statement index in your case?